### ISSUE TITLE
 Resource exhausted using data generator into model.fit

### ISSUE THREAD

**Issue by marlon-shiftone** [keras-team/keras#14559]
I'm training a neural network with multiple files:

   ```
 In: list_files
    
    Out: ['0370.npy', '0395.npy', ..., '9668.npy']
 

    In: len(list_files)

    Out: 22
```

Each file contains a list of 100k matrixes, like images. If I pick just one or two files, the model trains fine. 

In order to train my model I'm using a keras generator:

    ```
class DataGenerator(keras.utils.Sequence):
        'Generates data for Keras'
        def __init__(self, list_IDs, batch_size=96, shuffle=True):
            'Initialization'
            self.batch_size = batch_size
            self.list_IDs = list_IDs
            self.shuffle = shuffle
            self.on_epoch_end()
    
        def __len__(self):
            'Denotes the number of batches per epoch'
            #return int(np.floor(len(self.list_IDs) / self.batch_size))
            return 20
        
        def __getitem__(self, index):
            'Generate one batch of data'
            # Generate indexes of the batch
            print(index)
            indexes = self.indexes[index]
    
            # Find list of IDs
         
            list_IDs_temp = self.list_IDs[indexes]
    
            # Generate data
            X, y = self.__data_generation(list_IDs_temp)
    
            return X, y
    
        def on_epoch_end(self):
            'Updates indexes after each epoch'
            self.indexes = np.arange(len(self.list_IDs))
            if self.shuffle == True:
                np.random.shuffle(self.indexes)
    
        def __data_generation(self, list_IDs_temp):
            'Generates data containing batch_size samples' # X : (n_samples, *dim, n_channels)
            # Initialization
            X = np.empty((self.batch_size, 10, 10, 20))
            y = np.empty((1,), dtype=float)
    
            # Generate data
            df = np.load(list_IDs_temp)
            input = df[:, 0:-1]
            labels = df[:, -1]
            labels = labels.reshape(-1, 1)
            min_max_scaler = MinMaxScaler(feature_range=(0, 1))
            input = min_max_scaler.fit_transform(input)
            dataset = np.concatenate([input,labels], axis=1)
            input, labels = f.separar_multi(dataset, n_steps=100)
            labels = labels.reshape(-1, 1)
            input = f.separar_uni_X(input, n_passos=96)
            labels = f.separar_uni_y(labels, n_passos=96)
            input = input.reshape(-1, 96, 10, 10, input.shape[-1])
            return input, labels
            K.clear_session()
```

However, I'm still get the error:
```

    (0) Resource exhausted: OOM when allocating tensor with shape[1632864,64,10,10] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
             [[{{node time_distributed_1/convolution}}]]
    Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
    
             [[metrics/kW_perdidos/Identity/_165]]
    Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
    
      (1) Resource exhausted: OOM when allocating tensor with shape[1632864,64,10,10] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
             [[{{node time_distributed_1/convolution}}]]
    Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
    
    0 successful operations.
    0 derived errors ignored.

```
The generator wasn't supposed to pick each file separately and feed it into the neural network to avoid this problem?

Any suggestion is welcomed!

**Comment by bguetarni** [comment]
The problem is not in the RAM but rather in the VRAM. It's the memory of your GPU that is exhausted. When you use a GPU, the CPU prepares a batch of input/outputs and send them to your GPU with the operations to compute (model inference, loss computations and some others). Here it seems that your GPU memory is exhausted because of too much things to store.

To overcome this you can reduce the batch size or use a smaller dtype (for example if your images dtype is _float64_, try with _float32_).

**Comment by sushreebarsa** [comment]
@marlon-shiftone Could you please refer to the above comment and let us know if it helps ? Thanks!

**Comment by marlon-shiftone** [comment]
> @marlon-shiftone Could you please refer to the above comment and let us know if it helps ? Thanks!

Thanks for the comment. That is right!

**Comment by sushreebarsa** [comment]
@marlon-shiftone Thanks for the update! Can you please let us know if we can close this issue as it has been resolved? Thank you!

**Comment by marlon-shiftone** [comment]
> @marlon-shiftone Thanks for the update! Can you please let us know if we can close this issue as it has been resolved? Thank you!

Yes please!
